### PR TITLE
Fix identify2 fast cache, other optimizations

### PR DESCRIPTION
**Pull request by patrickxb**
1. identify2 fast cache was broken, CachedAt was an invalid time and thus always timed out
2. added cache of self's userpluskeys, but I don't think it is useful.  Never gets hit.  Happy to delete it.  Perhaps it is useful for rekey.
3. short-circuit MerkleClient.LoadRoot() if already loaded

r? @maxtaco

### CODE REVIEW

File: go/libkb/identify2_cache.go [hunk 1]
```diff
@@ -70,7 +70,7 @@ func (c *Identify2Cache) Get(uid keybase1.UID, gctf GetCheckTimeFunc, timeout ti
 func (c *Identify2Cache) Insert(up *keybase1.UserPlusKeys) error {
 	tmp := *up
 	copy := &tmp
-	copy.Uvv.CachedAt = keybase1.Time(time.Now().Unix())
+	copy.Uvv.CachedAt = keybase1.ToTime(time.Now())
```
Comment (patrickxb) [inline go/libkb/identify2_cache.go]: This is the main fix.

File: go/engine/resolve_identify2.go [hunk 2]
```diff
@@ -71,6 +71,11 @@ func (e *ResolveThenIdentify2) resolveUID(ctx *Context) (err error) {
 		return err
 	}
 	e.arg.Uid = rres.GetUID()
+	if rres.WasKbUsernameAssertion() && !e.arg.NeedProofSet {
+		// the resolve assertion was a keybase username, so remove it
+		// from identify2 arg to allow cache hits on UID.
+		e.arg.UserAssertion = ""
```
Comment (patrickxb) [inline go/engine/resolve_identify2.go]: This is also needed in order for the fast cache to ever be used.

Comment (maxtaco) [inline go/engine/resolve_identify2.go]: what about By-UID assertions?

**Comment by maxtaco** [comment]
👍

File: go/engine/resolve_identify2.go (hunk 2 above)
Comment (patrickxb) [inline go/engine/resolve_identify2.go]: added a change to handle those too

**Comment by patrickxb** [comment]
Do you want me to leave userpluskeys cache in or remove it?  

Also, did you see merkle client change?  I'm cool removing that, too.  Not a big performance boost.

File: go/libkb/merkle_client.go [hunk 3]
```diff
@@ -168,6 +168,13 @@ func merkleHeadKey() DbKey {
 
 func (mc *MerkleClient) LoadRoot() error {
 	mc.G().Log.Debug("+ MerkleClient.LoadRoot()")
+	mc.RLock()
+	if mc.lastRoot != nil {
```
Comment (maxtaco) [inline go/libkb/merkle_client.go]: when is this value reset?

Comment (patrickxb) [inline go/libkb/merkle_client.go]: only in this function

Comment (maxtaco) [inline go/libkb/merkle_client.go]: i'm worried we're going to start getting stale results then...  i'd say back this out...

Comment (patrickxb) [inline go/libkb/merkle_client.go]: Sure thing.  I was worried too, but this function showed up during profiling...

**Comment by patrickxb** [comment]
@maxtaco I'm going to disable the UserPlusKeys cache for safety.  We can enable later if we feel necessary for Rekey, but I think this id2 fix will be significant perf boost.  Might not need it.

**Comment by patrickxb** [comment]
retest this please

**Pull request merged**